### PR TITLE
[RW-1119][Risk=No] Add workspace list page action items

### PR DESCRIPTION
**Pull request by blrubenstein**
Shepherding this through for Greg. This adds the menu items to the workspace list. This will probably need specific tests, but want to open for review on basic functionality.
Please remember to keep in mind the security levels outlined in [CONTRIBUTING.md](CONTRIBUTING.md) and to 
include a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title

* **no**: None 
* **low**: Low chance of potential impact to, or exposure of patient data
* **moderate**: Moderate chance of potential impact to, or exposure of patient data
* **severe**: Severe chance of potential impact to, or exposure of patient data

Please also:

* Get thumbs from reviewer(s)
* Verify all tests go green, including CI tests

### CODE REVIEW

File: ui/src/app/views/workspace-list/component.ts [hunk 1]
```diff
@@ -58,19 +76,7 @@ export class WorkspaceListComponent implements OnInit, OnDestroy {
           // - While the billing project is being initialized, we want to keep the
           //   big spinner on the page to provide obvious messaging to the user
           //   about the expected wait time.
```
Comment (UrsaStutsman) [inline ui/src/app/views/workspace-list/component.ts]: May want to move these comments to the new function.

File: ui/src/app/views/workspace-share/component.spec.ts [hunk 2]
```diff
@@ -130,8 +140,11 @@ describe('WorkspaceShareComponent', () => {
   it('removes users correctly and does not allow self removal', fakeAsync(() => {
     workspaceSharePage.fixture.componentRef.instance.userEmail
         = 'sampleuser1@fake-research-aou.org';
+    workspaceSharePage.fixture.componentRef.instance.accessLevel
+        = WorkspaceAccessLevel.OWNER;
     workspaceSharePage.readPageData();
     const de = workspaceSharePage.fixture.debugElement;
+    console.log(de.queryAll(By.css('.user')));
```
Comment (UrsaStutsman) [inline ui/src/app/views/workspace-share/component.spec.ts]: Did you mean to leave this?

File: ui/src/app/views/workspace-edit/component.ts [hunk 3]
```diff
@@ -101,7 +101,7 @@ export class WorkspaceEditComponent implements OnInit {
   workspaceCreationConflictError = false;
   workspaceUpdateError = false;
   workspaceUpdateConflictError = false;
-  private accessLevel: WorkspaceAccessLevel;
+  accessLevel: WorkspaceAccessLevel;
```
Comment (UrsaStutsman) [inline ui/src/app/views/workspace-edit/component.ts]: Sorry! I totally let this discussion get derailed, was this where you landed? Best option?

File: ui/src/app/utils/workspacePermissions.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import {Workspace, WorkspaceAccessLevel, WorkspaceResponse} from '../../generated';
```
Comment (UrsaStutsman) [inline ui/src/app/utils/workspacePermissions.ts]: Excellent!

File: ui/src/app/views/workspace-list/component.css [hunk 5]
```diff
@@ -112,4 +106,64 @@
   width: 32px;
   height: 32px;
   color: #216FB4;
-}
\ No newline at end of file
+}
+
+.header-area {
```
Comment (UrsaStutsman) [inline ui/src/app/views/workspace-list/component.css]: More question than comment. Did you see the same issues on these cards as we saw on notebook and cohort card headers? If so, I just want to add a comment to that ticket, noting it's also present here.

**Comment by blrubenstein** [comment]
@UrsaStutsman should be good for another look.

File: ui/src/app/utils/workspacePermissions.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import {Workspace, WorkspaceAccessLevel, WorkspaceResponse} from 'generated';
+
+/**
+ * Composite class with helper functions useful in the UI.
```
Comment (dmohs) [inline ui/src/app/utils/workspacePermissions.ts]: Use your own judgement, but my gut says this information is obvious from the implementation. Nice helper class, though.

File: ui/src/app/views/workspace-list/component.html [hunk 7]
```diff
@@ -36,33 +36,93 @@ <h3 class="table-header-descriptor">Your Workspaces</h3>
...
+            <button class="action-item clone-item"
+                    routerLink="/workspaces/{{workspacePermissions.workspace.namespace}}/{{workspacePermissions.workspace.id}}/clone"
+                    routerLinkActive="active">
+              <clr-icon shape="copy"></clr-icon>
+              Clone
+            </button>
+            <span role="tooltip" aria-haspopup="true" class="tooltip tooltip-sm tooltip-top-right">
+              <button class="action-item edit-item"
+                      [disabled]="!workspacePermissions.canWrite"
+                      [class.disabled]="!workspacePermissions.canWrite"
```
Comment (dmohs) [inline ui/src/app/views/workspace-list/component.html]: Your Angular-foo is becoming quite formidable.

File: ui/src/app/views/workspace-list/component.spec.ts [hunk 8]
```diff
@@ -72,4 +115,69 @@ describe('WorkspaceListComponent', () => {
...
+    expect(workspaceListPage.fixture.componentInstance.deleteModal.deleting).toBeTruthy();
+    expect(workspaceListPage.fixture.componentInstance.deleteModal.resource)
+      .toEqual(firstWorkspace);
+    const deleteModal =
+      workspaceListPage.fixture.debugElement.query(By.css('app-confirm-delete-modal'));
+    simulateClick(workspaceListPage.fixture, deleteModal.query(By.css('.confirm-delete-btn')));
+    updateAndTick(workspaceListPage.fixture);
+    expect(deleteSpy).toHaveBeenCalledWith(firstWorkspace.namespace, firstWorkspace.id);
+    const numNewWorkspaces = workspaceListPage.fixture.componentInstance.workspaceList.length;
+    expect(numNewWorkspaces).toEqual(numWorkspaces - 1);
```
Comment (dmohs) [inline ui/src/app/views/workspace-list/component.spec.ts]: Wow. This entire test reads like a behavior specification. Beautiful.

File: ui/src/app/views/workspace-list/component.html (hunk 7 above)
Comment (blrubenstein) [reply ui/src/app/views/workspace-list/component.html]: This was actually Greg!

File: ui/src/app/utils/workspacePermissions.ts (hunk 6 above)
Comment (blrubenstein) [reply ui/src/app/utils/workspacePermissions.ts]: Also actually Greg!

**Pull request merged**